autofs-5.1.8 - bailout on rpc systemerror

From: Ian Kent <raven@themaw.net>

If there's a system error (eg. oversize packet received) just give up
since redoing the call would likely end up with the same error.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG      |    1 +
 lib/rpc_subs.c |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index b4b064ff..575f186d 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -7,6 +7,7 @@
 - fix nonstrict fail handling of last offset mount.
 - dont fail on duplicate offset entry tree add.
 - fix loop under run in cache_get_offset_parent().
+- bailout on rpc systemerror.
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
diff --git a/lib/rpc_subs.c b/lib/rpc_subs.c
index 7b8162b4..ee7f94b9 100644
--- a/lib/rpc_subs.c
+++ b/lib/rpc_subs.c
@@ -1195,6 +1195,8 @@ static int rpc_get_exports_proto(struct conn_info *info, struct exportinfo **exp
 				 info->timeout);
 		if (status == RPC_SUCCESS)
 			break;
+		if (status == RPC_SYSTEMERROR)
+			break;
 		if (++vers_entry > 2)
 			break;
 		CLNT_CONTROL(client, CLSET_VERS,
